launcher/src/main/java/com/skcraft/launcher/launch/JavaRuntimeFinder.java

```diff
@@ -75,6 +79,24 @@ private JavaRuntimeFinder() {
                     }
                 }).distinct().forEach(file -> entries.add(getRuntimeFromPath(file.getAbsolutePath())));
             }
+        } else if (env.getPlatform() == Platform.MAC_OS_X) {
+            launcherDir = new File(System.getenv("HOME"), "Library/Application Support/minecraft");
+
+            try {
+                Process p = Runtime.getRuntime().exec("/usr/libexec/java_home -X");
+                NSArray root = (NSArray) PropertyListParser.parse(p.getInputStream());
+                NSObject[] arr = root.getArray();
+                for (NSObject obj : arr) {
+                    NSDictionary dict = (NSDictionary) obj;
+                    entries.add(new JavaRuntime(
+                        new File(dict.objectForKey("JVMHomePath").toString()).getAbsoluteFile(),
+                        dict.objectForKey("JVMVersion").toString(),
+                        dict.objectForKey("JVMArch").toString().equals("x86_64")
```

Comment:
Specifically on Mac we may need to worry about the architecture being arm64?

Reply:
We would need to update the signature of that class to be an ENUM for arch instead of a boolean then.

Reply:
well, arm64 is still 64-bit. the only purpose of that field is to rank 64-bit installations over 32-bit, since 32-bit has memory limits

Reply:
A quick google search shows that for arm the 64-bit architecture is labeled aarch64. Can you fact check me on that.

Reply:
yep, that's the name of the architecture and it appears to be what Java uses as well

Reply:
Ok I'm currently refactoring the function